src/Compilers/Core/Portable/SourceGeneration/GeneratorAdaptor.cs

```diff
@@ -31,15 +32,56 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
             GeneratorInitializationContext oldContext = new GeneratorInitializationContext(context.CancellationToken);
             SourceGenerator.Initialize(oldContext);
 
-            if (oldContext.InfoBuilder.PostInitCallback is object)
-            {
-                context.RegisterForPostInitialization(oldContext.InfoBuilder.PostInitCallback);
-            }
+            context.InfoBuilder.PostInitCallback = oldContext.InfoBuilder.PostInitCallback;
+            context.InfoBuilder.SyntaxContextReceiverCreator = oldContext.InfoBuilder.SyntaxContextReceiverCreator;
 
             context.RegisterExecutionPipeline((ctx) =>
             {
-                // PROTOTYPE(source-generators): this is where we'll build the actual emulation pipeline
+                var context = ctx.Sources.Compilation
+                                         .Transform(c => new GeneratorContextBuilder(c))
+                                         .Join(ctx.Sources.ParseOptions).Transform(p => p.Item1 with { ParseOptions = p.Item2.FirstOrDefault() })
+                                         .Join(ctx.Sources.AnalyzerConfigOptions).Transform(p => p.Item1 with { ConfigOptions = p.Item2.FirstOrDefault() })
+                                         .Join(ctx.Sources.CreateSyntaxReceiver()).Transform(p => p.Item1 with { Receiver = p.Item2.FirstOrDefault() })
```

Comment:
I now more fully understand the flow here for the `InputNode<ISyntaxReceiver>`. As constructed in this PR the process seems very indirect. There is a builder threaded through to hold the state. The builder seems setup to handle the state being or not being created. Inside here we unconditionally create the state though. Not sure if in the future this gets more conditional or not. 

Was wondering why didn't you take approach of say having `SourceGeneratorAdapter` just proactively create both the `InputNode<ISyntaxReceiver>` and `IncrementalValueSource<ISyntaxReceiver>`. That could essentially be properties on the type. Then in the generator driver those could be used to seed the initial state table entries for the `ISyntaxReceiver`. 

Is there something I'm missing here that makes this more complex than I'm imagining.?

Reply:
Talked a bit offline about this, but essentially this is just setting up the groundwork for other per-generator sources such as syntax filters that occur outside the regular graph flow.